lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/RootContextLauncher.cs

```diff
@@ -40,15 +41,16 @@ internal sealed class RootContextLauncher
         private ContextRuntime _rootContext = null;
 
         public RootContextLauncher(string id, IConfiguration contextConfiguration,
-            Optional<ServiceConfiguration> rootServiceConfig, Optional<TaskConfiguration> rootTaskConfig)
+            Optional<ServiceConfiguration> rootServiceConfig, Optional<IConfiguration> rootTaskConfig, IHeartBeatManager heartbeatManager)
         {
             Id = id;
             _rootContextConfiguration = contextConfiguration;
             _rootServiceInjector = InjectServices(rootServiceConfig);
+            _rootServiceInjector.BindVolatileInstance(GenericType<IHeartBeatManager>.Class, heartbeatManager);
```

Comment:
We shouldn't create a brand new injector for the root service. If it is forked from evaluator injector, then heartbeatManager can be shared without using BindVolatileInstance.